### ISSUE TITLE
Touch Does Not Fire Off Click Events On Touch

### ISSUE THREAD

**Issue by tabrumle** [microsoft/fluentui#13155]
- **Package version(s)**: All
- **Browser and OS versions**: All

### Describe the issue:

When you touch a button it doesn't always fire off its execute handler. Specifically it won't fire for two cases I've noticed. 
1. On touch if there is any sort of touch movement is detected while your finger is down. This can make it near impossible for some users to execute buttons if they aren't able to keep their fingers steady and in the same place for the duration of the touch event.

2. There is also an issue where you aren't able to touch execute a button if you quickly tap it after it renders. We are seeing this with buttons within callouts that are rendered in response to a user's action.

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/The_Golden_Goose/pen/rNOrRXg

#### Actual behavior:

Button gets hover styling only but no execute behavior occurs.

#### Expected behavior:

Button should be executed just like if you mouse down on the button and mouse up within the same button.

**Comment by DanielBub** [comment]
The onClick handler of the BaseButton is not being called for touch tap in the first few seconds after the button is shown. 
That's also true for the click on flyouts.  

![FirstTapFloatie](https://user-images.githubusercontent.com/13901708/81927827-f98fe900-95ec-11ea-8e53-503047fd9269.gif)

![FirstTap](https://user-images.githubusercontent.com/13901708/81927561-84bcaf00-95ec-11ea-8080-6fbf396bd27b.gif)

**Comment by aneeshack4** [comment]
@tabrumle @DanielBub Thank you for reaching out to us. I'm not able to recreate the slight movement you're describing to repro this however I can imagine that this could be an accessibility problem. I appreciate y'all digging a bit and providing more information too. Going to tag Button's owner @khmakoto - is there anything with the onClick handlers we can improve to make this experience better?

**Comment by tabrumle** [comment]
Hey @aneeshack4 I updated the title and description to be offer a little more insight to the issue we are seeing as well as updated the code pen to showcase the issue we are seeing with  #2 in the description

**Comment by tabrumle** [comment]
@khmakoto friendly ping on this. This issue is currently a blocker for us.

**Comment by ecraig12345** [comment]
For the first issue, I'm not sure this is something we can/should try to fix, since we're already handling clicks (which should also catch complete taps) and it's up to the OS to determine what's a click/tap vs drag. If you're seeing a particularly severe issue on a particular OS/browser with things that should register as taps registering as drags, that should be reported to the OS and/or browser vendor, not us.

For the second issue, I wasn't able to repro this on iOS Safari or Chrome (and don't have any other touchscreen devices available to test). Which device/browser are you seeing the issue in, so I can ask another team member to try and repro? Though fair warning that I'm also not sure this will be something we can/should fix on our side, even if we get a repro.
